### PR TITLE
Fix get avatar authorization

### PR DESCRIPTION
**Pull request by Pytal**
This is functionally equivalent but explicitly handles all scopes and is slightly more restrictive by returning a placeholder in the single case where a `PropertyDoesNotExistException` is thrown when querying for the avatar account property

### To Do

- [x] Add new test cases
- [x] Fix tests

### CODE REVIEW

**Review by blizzz — APPROVED** [review]
Perhaps some tests though?

File: lib/private/Avatar/AvatarManager.php [hunk 1]
```diff
@@ -136,20 +136,25 @@ public function getAvatar(string $userId) : IAvatar {
...
-			return new PlaceholderAvatar($folder, $user, $this->logger);
+		switch ($avatarScope) {
+			case IAccountManager::SCOPE_PRIVATE:
+				if ($requestingUser !== null && $this->knownUserService->isKnownToUser($requestingUser->getUID(), $userId)) {
+					return new UserAvatar($folder, $this->l, $user, $this->logger, $this->config);
+				}
+				break;
+			case IAccountManager::SCOPE_LOCAL:
+			case IAccountManager::SCOPE_FEDERATED:
+				if ($requestingUser !== null) {
```
Comment (juliushaertl) [inline lib/private/Avatar/AvatarManager.php:147]: This doesn't match with how the `SCOPE_LOCAL` was working before:

https://github.com/nextcloud/server/blob/4d26a9afa01aaf069e29a62f4e9547c34156ea01/lib/public/Accounts/IAccountManager.php#L51

Is that an intended change? If so maybe you can elaborate a bit about the reasoning :)

**Review by juliusknorr — CHANGES_REQUESTED** [review]

File: lib/private/Avatar/AvatarManager.php (hunk 1 above)
Comment (Pytal) [reply lib/private/Avatar/AvatarManager.php]: Fixed it! The changes before were based on semantics i.e. "🔒 Local" seemed like it'd be more restrictive

File: lib/private/Avatar/AvatarManager.php [hunk 2]
```diff
@@ -136,20 +136,21 @@ public function getAvatar(string $userId) : IAvatar {
...
+			case IAccountManager::SCOPE_LOCAL:
+			case IAccountManager::SCOPE_FEDERATED:
+			case IAccountManager::SCOPE_PUBLISHED:
+				return new UserAvatar($folder, $this->l, $user, $this->logger, $this->config);
+			default:
+				return new PlaceholderAvatar($folder, $user, $this->logger);
 		}
 
-		return new UserAvatar($folder, $this->l, $user, $this->logger, $this->config);
+		return new PlaceholderAvatar($folder, $user, $this->logger);
```
Comment (artonge) [inline lib/private/Avatar/AvatarManager.php:153]: Unreachable code, no ?

File: lib/private/Avatar/AvatarManager.php [hunk 3]
```diff
@@ -136,20 +136,21 @@ public function getAvatar(string $userId) : IAvatar {
...
-				$requestingUser === null
-				// logged in, but unknown to user
-				|| !$this->knownUserService->isKnownToUser($requestingUser->getUID(), $userId)
-			)) {
-			// use a placeholder avatar which caches the generated images
-			return new PlaceholderAvatar($folder, $user, $this->logger);
+		switch ($avatarScope) {
+			case IAccountManager::SCOPE_PRIVATE:
+				if ($requestingUser !== null && $this->knownUserService->isKnownToUser($requestingUser->getUID(), $userId)) {
+					return new UserAvatar($folder, $this->l, $user, $this->logger, $this->config);
```
Comment (artonge) [inline lib/private/Avatar/AvatarManager.php:142]: I would keep the previous comments if they are still valid.

File: lib/private/Avatar/AvatarManager.php (hunk 2 above)
Comment (Pytal) [reply lib/private/Avatar/AvatarManager.php:153]: Will be hit after the break in the switch statement above i.e. when the requesting user is not logged in

File: lib/private/Avatar/AvatarManager.php (hunk 3 above)
Comment (Pytal) [reply lib/private/Avatar/AvatarManager.php:143]: Added back comment explanations not already expressed clearly by the code

**Review by artonge — APPROVED** [review]

**Comment by Pytal** [comment]
CI failure unrealated

**Comment by Pytal** [comment]
CI failure unrelated

**Pull request merged**